### PR TITLE
Mesh-of-edges feature

### PR DESCRIPTION
**Pull request by ted-ross**
This feature creates a new connection role called inter-edge.  This can be used to create full-meshes of edge routers that may be connected to an interior network of routers.  The main benefit is that edge sites in a network can be scaled up to include multiple routers without a) requiring inter-edge traffic to flow via the interior or b) eating into the budget of interior routers (128 max) because you had to use interior routers at the edge.

In an edge-mesh, all local traffic flows between the edge routers directly (one hop maximum).  

[Mesh of Edges Notes.pdf](https://github.com/skupperproject/skupper-router/files/9982028/Mesh.of.Edges.Notes.pdf)

### CODE REVIEW

**Review by kgiusti — APPROVED** [review]
LGTM.
Some in-line nits and comment requests for your consideration but nothing to hold up landing this.  Nice work!

File: src/router_core/forwarder.c [hunk 1]
```diff
@@ -444,8 +455,21 @@ static uint8_t qdr_forward_effective_priority(qd_message_t *msg, qdr_address_t *
...
+    }
+
+    if (out_link->conn->role == QDR_ROLE_EDGE_CONNECTION && out_link->conn->edge_mesh_id[0] != '\0') {
+        qd_parsed_field_t *mesh_id = qd_message_get_ingress_mesh(in_dlv->msg);
+        mesh_loop = !!mesh_id && qd_iterator_equal_n(qd_parse_raw(mesh_id), (unsigned char*) out_link->conn->edge_mesh_id, QD_DISCRIMINATOR_BYTES);
+    }
+
+    return (((in_dlv->via_edge && in_link->conn == out_link->conn)
+              || mesh_loop
+              || ((in_link->conn->role == QDR_ROLE_INTER_EDGE || in_link->conn->role == QDR_ROLE_EDGE_CONNECTION) && out_link->proxy)));
```
[the remark this answers is not in the document]
Comment (ted-ross) [reply src/router_core/forwarder.c:472]: I added comments to the boolean expressions in that function.

File: tests/interrouter_msg.py [hunk 2]
```diff
@@ -99,7 +99,8 @@ def decode(data: bytes) -> 'RouterAnnotationsSection':
         obj = Data()
         obj.decode(data)
         lcount = obj.get_list()
-        assert lcount == 4
+        print("lcount == %d" % lcount)
```
[the remark this answers is not in the document]
Comment (ted-ross) [reply tests/interrouter_msg.py:102]: Yes, thank you.

File: src/router_core/modules/mobile_sync/mobile.c [hunk 3]
```diff
@@ -49,18 +49,19 @@ static const char *HAVE_SEQ   = "have_seq";
...
 //
 // qdr_node_t.sync_mask bit values
 //
 #define ADDR_SYNC_ROUTER_MA_REQUESTED     0x00000001
 #define ADDR_SYNC_ROUTER_VERSION_LOGGED   0x00000002
 
-#define BIT_SET(M,B)   M |= B
-#define BIT_CLEAR(M,B) M &= ~B
-#define BIT_IS_SET(M,B) (M & B)
+#define BIT_SET(M,B)    (M) |= (B)
```
[the remark this answers is not in the document]
Comment (ted-ross) [reply src/router_core/modules/mobile_sync/mobile.c:62]: Done.

**Pull request merged**